go/cmd/dolt/cli/stdio.go

```diff
@@ -141,6 +142,66 @@ func PrintErrf(format string, a ...interface{}) {
 	fmt.Fprintf(CliErr, format, a...)
 }
 
+// EphemeralPrinter is tool than you can use to print temporary line(s) to the
+// console. Each time Display is called, the output is reset, and you can begin
+// writing anew. If you need to display multiple temporary lines, call Newline,
+// before writing each line.
+type EphemeralPrinter struct {
+	outW io.Writer
+	w    *uilive.Writer
+}
+
+// StartEphemeralPrinter creates a new EphemeralPrinter and starts it. You
+// should defer Stop after calling this.
+func StartEphemeralPrinter() *EphemeralPrinter {
+	w := uilive.New()
+	w.Out = CliOut
+	e := &EphemeralPrinter{w, w}
+	e.start()
+	return e
+}
+
+// Printf formats and prints a string to the printer. You should not add
+// newlines in this string. If you need to display multiple lines, use Newline.
```

Comment:
How important is it to not add newlines? Does it screw up the display and line clearing? If so... you might want to consider returning an error or panic'ing if the caller has passed in a newline as an extra protection against than happening, instead of relying on people to read and follow the comments.

Reply:
Or... is it possible to detect the newline and automatically handle it correctly? i.e. by hiding the current Newline method so callers don't ever have to learn that interface?

Reply:
Continuing that thought about simplifying the caller's interface... is it helpful to combine the Printf method and Display? It seems like the calling code typically calls Printf, then (almost always) immediately calls Display. A small change, but might make the interface just a little bit easier to use if callers don't have to know there is a two step process to update text on screen.

Reply:
It does screw up the display and line clearing! Good points, let me  add this.